### ISSUE TITLE
Compiling from source for macos: upgrade onig version

### ISSUE THREAD

**Issue by happysalada** [vectordotdev/vector#9710]
### Vector Version

```
0.17.2
```

### Vector Configuration File

N/A

### Debug Output

```
error[E0425]: cannot find value `OnigSyntaxPython` in crate `onig_sys`
  --> /private/tmp/nix-build-vector-0.17.2.drv-0/vector-0.17.2-vendor.tar.gz/onig/src/syntax.rs:39:39
   |
39 |         unsafe { transmute(&onig_sys::OnigSyntaxPython) }
   |                                       ^^^^^^^^^^^^^^^^ help: a static with a similar name exists: `On
igSyntaxPerl`
   |
  ::: /private/tmp/nix-build-vector-0.17.2.drv-0/source/target/x86_64-apple-darwin/release/build/onig_sys-
748c6a617139d063/out/bindings.rs:3:22644
   |
3  | ... } extern "C" { pub static mut OnigSyntaxPerl : OnigSyntaxType ; } extern "C" { pub static mut Oni
gSyntaxPerl_NG : OnigSyntaxType ; } ...
   |                    ------------------------------------------------ similarly named static `OnigSynta
xPerl` defined here
```

### Expected Behavior

compiling

### Actual Behavior

compile error

### Example Data

N/A

### Additional Context

I'm having trouble compiling from source on macos. It seems that the version of onig in the lock file is a (6.2.0) is not one that appears on the github releases
https://github.com/rust-onig/rust-onig/releases

Would it be possible to bump the onig version to (6.3.0) ?

### References

https://github.com/vectordotdev/vector/blob/master/Cargo.lock#L4172

**Comment by jszwedko** [comment]
Thanks @happysalada ! Can you confirm that you have the correct version of `onig_sys` vendored? It should be `69.7.0`.

**Comment by happysalada** [comment]
Hey, thank you very much for the quick turnaround!
The bad news is that I've tested again on master and the bug is still here.
The good news is that I think I've found the source.
The build I use is vendoring all the dependencies. The onig dependency has a dependency on the wrong version of onig_sys, I've filed a bug upstream https://github.com/rust-onig/rust-onig/issues/165
I think this can safely be closed, since it has nothing to do with vector.
Thank you again for the light-speed handling of this!
